src/generic_core/globals.ts

```diff
@@ -56,6 +56,7 @@ export var settings :uproxy_core_api.GlobalSettings = {
   proxyBypass: DEFAULT_PROXY_BYPASS.slice(0),
   enforceProxyServerValidity: false,
   validProxyServers: [],
+  promo: null,
```

Comment:
I would have expected this to be the promo code itself but elsewhere you set it to `off2016` - worth a one-liner explaining what this is.